### DISCUSSION TITLE
Passing no parameter to `withoutName()` exclude all the class

### DISCUSSION THREAD

**Discussion by bmarty** [LemonAppDev/konsist#1112] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.15.1

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Not a big deal and should not happen IRL, but using `withoutName()` without given names actually exclude all the classes. I expected that it excluded nothing.

In other words:
```
    @Test
    fun `demo that withoutName without param exclude all the classes`() {
        Konsist.scopeFromProject()
            .classes()
            .withoutName()
            .assertEmpty()
    }
```

is passing and I expected it to fail.

There is maybe the same issue on other scope where `withoutName` is available or even for other methods like `withoutClassNamed`.

### Actual behaviour

`.withoutName()` excludes all the classes

### Expected behaviour

`.withoutName()` do not exclude any classes

**Comment by nataliapeterwas** [comment, marked as answer]
The current behaviour of `withoutName` is intentional, as it only returns declarations that do not have a name. Since every class has a name, `withoutName` excludes all of them and returns an empty list. The opposite of this method is `withName`, which will return list of all classes, as they all have a name.

We are aware that this is somewhat of an edge case, but the method was added for consistency with other parts of the code — there are always two extensions: `withX` and `withoutX`. The intention behind this method is that it will be used as `withoutName("someName")`, to exclude declarations that have a specific name.
